contracts/core/EntryPoint.sol

```diff
@@ -357,7 +360,11 @@ contract EntryPoint is IEntryPoint, StakeManager {
         }
         paymasterInfo.deposit = uint112(deposit - requiredPreFund);
         uint256 gas = mUserOp.verificationGasLimit - gasUsedByValidateWalletPrepayment;
-        try IPaymaster(paymaster).validatePaymasterUserOp{gas : gas}(op, opInfo.requestId, requiredPreFund) returns (bytes memory _context){
+        try IPaymaster(paymaster).validatePaymasterUserOp{gas : gas}(op, opInfo.requestId, requiredPreFund) returns (bytes memory _context, uint256 deadline){
+            // solhint-disable-next-line not-rely-on-time
+            if (deadline != 0 && deadline < block.timestamp) {
+                revert FailedOp(opIndex, address(0), "expired");
```

Comment:
Any reason to use address(0) in this case and not attribute the expiry to the paymaster if it was a paymaster expiry?  A paymaster could abuse it to invalidate many ops by returning an inconsistent value based on a storage slot, just like other invalidations by a paymaster.

Reply:
it was. but then I copy-and-pasted the line...